src/fmu/dataio/tools/rms/volumetrics.py

```diff
@@ -0,0 +1,228 @@
+            content="volumes",
+            unit="m3" if self._units == "metric" else "ft3",
+            vertical_domain={"depth": "msl"},
+            workflow=self.workflow,
+            forcefolder=self.forcefolder,
+            classification=self.classification,
+            tagname=self.tagname,
+            name=self.gridname,
+        )
+
+        out = edata.export(self.dataframe)
+        _logger.debug("Volume result to: %s", out)
+        return {"volume_table": out}
+
+
+def export_volumetrics(
+    project: RmsProject,
+    gridname: str,
+    volumejob_name: str,
+    global_config: dict = field(default_factory=dict),
+    forcefolder: str = "",
+    tagname: str = "vol",
+    classification: str = "restricted",
+    workflow: str = "rms volumetrics",
+    include_zone_maps: bool = False,
+    include_total_maps: bool = False,
+    include_3dgrid_properties: bool = False,
+) -> dict[str, str]:
+    """A function for simplified 'one-liner' dataio export from RMS volumetrics.
+
+    As the export_volumetrics may have multiple output (storing both tables, maps and
+    3D grids), the output from this function is always a dictionary. The table is
+    mandatory output, while maps and 3D grid are optional (not yet implemented).
+
+    Args:
+        project: The 'magic' project variable in RMS.
+        gridname: Name of 3D grid model in RMS.
+        volumejob_name: Name of the volume job.
+        global_config: The global configuration (may be redundant in a near future?)
+        forcefolder: As default, volume tables will be exported to the file structure,
+            and the folder name will be 'tables'. This can be overriden here. For
+            optional assosiated volume maps and grids, the default folder names
+            cannot be changed.
+        tagname: Defaulted to 'vol' for this function
+        classification: Use 'internal' or 'restricted' (default).
+        workflow: Information about he work flow; defaulted to 'rms volumetrics'.
+        include_zone_maps: If True, then all the zone maps that are toggled on in the
+            volume job are exported also (currently NOT implemented).
+        include_total_maps: If True, then all the total maps that are toggled on in the
+            volume job are exported also (currently NOT implemented).
+        include_3dgrid_propererties: If True, then all the 3D properties (also called
+            parameters) that are toggled on in the volume job are exported also,
+            together with the 3D grid geometry (currently NOT implemented).
+    """
+
+    task = _RmsVolumetricsExport(
+        project,
+        gridname,
+        volumejob_name,
+        config=global_config,
+        forcefolder=forcefolder,
+        tagname=tagname,
+        workflow=workflow,
+        classification=classification,
+        include_zone_maps=include_zone_maps,
+        include_total_maps=include_total_maps,
+        include_3dgrid_properties=include_3dgrid_properties,
+    )
```

Comment:
I am considering dropping the function ans just the class public instead

Reply:
That is now done

Reply:
Maybe we should have a discussion on the kind of usage pattern we want to expose to users going forward. I think exposing a single function with just a call is a simpler usage pattern than exposing a class that is instantiated and has a method called on it. But, we might not be able to do that in all cases. In particular for this volumetrics functionality there is not really a need to use a class at all -- we could just as easily accomplish the same thing with private functions in the module and one public function doing the export

Reply:
Agree partly, and I am changing my mind every other week on this topic. Sometimes working with a class is simpler for the programmer; no need to transfer long lists of variables between functions. The `self` becomes a better way for sharing variables than Fortran's common blocks (yes I did that a lot in the previous millenium... not without pain though)  :-) 

The other reason for making it a class with currently  one export method is that in future this can be expanded to a few more public methods, like `RmsVolumetricsExporter(...).to_sumo()` or something like that, that will take the data directly to sumo in stead of to scratch disk.

Reply:
I think I would have preferred that `.export()` had a Sumo option which we could default to `False`, then eventually to `True`, but different discussion possibly...

The One Class Per Export pattern is similar to what was in `js-fmu-metadata` - the predecessor of `fmu-dataio`. If I remember correctly, we had a list of reasons why we didn't want to keep doing it that way back then. Not sure if those are still valid.

As for transferring `self`, will it not work to have a (wrapper) function initiating a class object, and then passing that to others? (Have not looked into any details, so this is skudd fra hoften.)

Arguments _against_ having the user initiating the class that I can think of:
- User now has a class in his script, and might decide to do something else with it, rather than just immediately calling `.export()`
- All the export functions that are _outside_ RMS will be in the form of FORWARD_MODELS, and will look more like function calls, than class initialization. So we risk to have different logic inside vs outside RMS, in the same FMU workflow. Perhaps it would be helpful if the first use case here was built both inside and outside RMS at the same time, so that we spot those logical differences early. Experience from early-days of fmu-dataio is that we started inside RMS, with the assumption that we were inside RMS, and then there was some overhead when that had to replicated _outside_ RMS.